client/components/language-picker/modal.jsx

```diff
@@ -0,0 +1,189 @@
+			filter: 'popular',
+			search: false,
+			selectedLanguageSlug: this.props.selected,
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.selected !== this.props.selected ) {
+			this.setState( {
+				selectedLanguageSlug: nextProps.selected
+			} );
+		}
+	}
+
+	getFilterLabel( filter ) {
+		const { translate } = this.props;
+
+		switch ( filter ) {
+			case 'popular':
+				return translate( 'Popular languages', { textOnly: true } );
+			default:
+				return translate( 'All languages', { textOnly: true } );
+		}
+	}
+
+	getDisplayedLanguages() {
+		const { languages } = this.props;
+		const { search, filter } = this.state;
+
+		if ( search ) {
+			const searchString = search.toLowerCase();
+			return languages.filter( language => {
+				return includes( language.name.toLowerCase(), searchString );
+			} );
+		}
+
+		switch ( filter ) {
+			case 'popular':
+				const popularLanguages = languages.filter( language => language.popular );
+				popularLanguages.sort( ( a, b ) => a.popular - b.popular );
+				return popularLanguages;
+			default:
+				return languages;
+		}
+	}
+
+	handleSearch = ( search ) => {
+		this.setState( { search } );
+	}
+
+	handleClick = ( selectedLanguageSlug ) => {
+		this.setState( { selectedLanguageSlug } );
+	}
+
+	handleSelectLanguage = () => {
+		const langSlug = this.state.selectedLanguageSlug;
+		this.props.onSelected( langSlug );
+		this.handleClose();
+	}
+
+	handleClose = () => {
+		this.props.onClose();
+	}
+
+	renderTabItems() {
+		const tabs = [ 'popular', '' ];
+
+		return map( tabs, filter => {
+			const selected = this.state.filter === filter;
+			const onClick = () => this.setState( { filter } );
+
+			return (
+				<SectionNavTabItem
+					key={ filter }
+					selected={ selected }
+					onClick={ onClick }
+				>
+					{ this.getFilterLabel( filter ) }
+				</SectionNavTabItem>
+			);
+		} );
+	}
+
+	renderLanguageList() {
+		const languages = this.getDisplayedLanguages();
+
+		return (
+			<div className="language-picker__modal-list">
+				{ map( languages, this.renderLanguageItem ) }
+			</div>
+		);
+	}
+
+	renderLanguageItem = ( language ) => {
+		const isSelected = language.langSlug === this.state.selectedLanguageSlug;
+		const classes = classNames( 'language-picker__modal-text', {
+			'is-selected': isSelected
+		} );
+
+		return (
+			<div
+				className="language-picker__modal-item"
+				key={ language.langSlug }
+				onClick={ partial( this.handleClick, language.langSlug ) }
+			>
+				<span className={ classes }>{ language.name }</span>
+			</div>
+		);
+	}
+
+	render() {
+		const { isVisible, translate } = this.props;
+
+		if ( ! isVisible ) {
+			// Render nothing at all if the modal is not visible
+			// <Dialog isVisible={ false }> still renders a lot of useless elements
+			return null;
+		}
+
+		const buttons = [
+			{
+				action: 'cancel',
+				label: translate( 'Cancel' )
+			},
+			{
+				action: 'confirm',
+				label: translate( 'Select Language' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **60** times:
`translate( 'Select a language below' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/language-picker)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).